lib/timers.js

```diff
@@ -477,47 +477,109 @@ exports.clearImmediate = function(immediate) {
 
 var unrefList, unrefTimer;
 
+function _makeTimerTimeout(timer) {
+  var domain = timer.domain;
+  var msecs = timer._idleTimeout;
+
+  // Timer has been unenrolled by another timer that fired at the same time,
+  // so don't make it timeout.
+  if (!msecs || msecs < 0)
+    return;
+
+  if (!timer._onTimeout)
+    return;
+
+  if (domain) {
+    if (domain._disposed)
+      return;
+
+    domain.enter();
+  }
+
+  debug('unreftimer firing timeout');
+  L.remove(timer);
```

Comment:
Shouldn't the timer be removed in case of bailout above as well?  E.g. when the domain is dead?